### PR TITLE
User management screen

### PR DESCRIPTION
**Pull request by abishekTa-egov**
User-management screen

## Release Notes

- **New Features**
  - Introduced new CSS classes for improved styling of the microplan interface.
  - Added a user management link to the MicroplanCard for easier navigation.
  - Enhanced user management functionalities with new configurations and methods.
  - Launched a new UserManagement component for managing user functionalities.

- **Bug Fixes**
  - Improved session management and navigation handling in the UserManagement component.

- **Chores**
  - Updated stylesheet links to the latest version for improved styling consistency.

### CODE REVIEW

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UICustomizations.js [hunk 1]
```diff
@@ -98,4 +101,130 @@ export const UICustomizations = {
...
+      const userInfo = Digit.UserService.getUser();
+      const tenantId = Digit.ULBService.getCurrentTenantId();
+      
+      return {
+        params: {},
+        url: '/mdms-v2/v2/_search', //mdms fetch from
+
+        body: {
+          MdmsCriteria: {
+            tenantId: "mz",
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UICustomizations.js:170]: hardcode tenantId

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
+export const tqmInboxConfig = {
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js:1]: name it properly

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  "tenantId": "pb",
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js:2]: mz tenant

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+  "moduleName": "UserManagementConfig",
+  "tqmInboxConfig": [
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js:4]: name it properly nowhere it should say tqm

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+          },
+          "children": {},
+          "show": true
+        },
+        "links": {
+          "uiConfig": {
+            "links": [
+              {
+                "text": "Bulk Upload Users",
+                "url": "/employee",
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js:117]: put propery urls for these links

Comment (nipunarora-eGov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js:117]: it should be under microplan/bulk-upload. something like this

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+              "component": "TqmInboxIcon",
+              "customClass": "inbox-links-icon"
+            }
+          },
+          "children": {},
+          "show": true
+        },
+        "filter": {
+          "uiConfig": {
+            "headerLabel": "TQM_INBOX_FILTER",
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js:141]: tqm?

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserSearchConfig.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
+const defaultSearchValues = {
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserSearchConfig.js:1]: what is this file?

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/UserManagement.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+const UserManagement = () => {
+    const { t } = useTranslation();
+    const location = useLocation()
+    const moduleName = Digit?.Utils?.getConfigModuleName() || "commonSanitationUiConfig"
+    const tenant = Digit.ULBService.getStateId();
+    
+
+    
+    const config = tqmInboxConfig?.tqmInboxConfig?.[0];
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/UserManagement.js:18]: nowhere it should say tqm

**Review by nipunarora-eGov — CHANGES_REQUESTED** [review]
@abishekTa-egov check out the comments and fix

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js (hunk 5 above)
Comment (abishekTa-egov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js:117]: I am not sure of the where to redirect

**Review by nipunarora-eGov — APPROVED** [review]

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+                  "MICROPLAN_ADMIN"
+                ]
+              },
+              {
+                "text": "Download User Data",
+                "url": "/employee",
+                "roles": [
+                  "MICROPLAN_ADMIN"
+                ]
+              }
```
[the remark this answers is not in the document]
Comment (nipunarora-eGov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js:128]: @abishekTa-egov resolve this add localisation key

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js (hunk 5 above)
Comment (nipunarora-eGov) [reply health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UserManagementConfig.js:117]: they will develop some screen for it just have a url like . /microplan-ui/bulk-upload-users. It should just redirect. that screen will be empty

**Review by nipunarora-eGov — APPROVED** [review]

**Review by nipunarora-eGov — APPROVED** [review]

**Pull request merged**